### DISCUSSION TITLE
qemu-system-aarch64 memory balloons to 20GB, even though explicitly set to 12GB

### DISCUSSION THREAD

**Discussion by jdolitsky** [lima-vm/lima#675] [Q&A]
### Description

In my lima config, I set `memory` explicitly to `12GiB`.

`limactl list` shows this:

```
NAME     STATUS     SSH                ARCH       CPUS    MEMORY    DISK      DIR
myenv    Running    127.0.0.1:59699    aarch64    6       12GiB     100GiB    /Users/jdolitsky/.lima/myenv
```

After running a bunch of containers in the VM, the mac "Activity Monitor" shows the memory of `qemu-system-aarch64` to be roughly 20 GB (see screenshot below):

<img width="793" alt="Screen Shot 2022-02-18 at 11 04 15 AM" src="https://user-images.githubusercontent.com/393494/154729189-8c2d9bbc-12e9-4ed0-9487-1e1b78acc27c.png">

This has been confirmed on 3 separate machines (all M1 Macs), running lima version 0.8.2

**Comment by jdolitsky** [comment]
Just tested on new release (0.8.3), and the issue still exists

**Comment by abiosoft** [comment]
@jdolitsky that is not the actual memory being used, right-click on the any of the column title and include `Real Memory` column.

**Comment by jdolitsky** [comment]
@abiosoft - thanks for sharing that. Found this answer which tries to explain the differences in the memory fields:
https://apple.stackexchange.com/a/107

Here's what I get with a `16GiB` lima VM, and all memory columns active:

<img width="912" alt="Screen Shot 2022-02-22 at 10 01 27 AM" src="https://user-images.githubusercontent.com/393494/155170900-a164e119-d045-43a6-9d18-d36994d4e119.png">

Still, none of those numbers appear to equal exactly 16, so I'm still not sure. I even considered the difference between `GiB` and `GB` (16 GiB = 17.179869184 GB).

**Reply by AkihiroSuda** [reply]
> Still, none of those numbers appear to equal exactly 16, so I'm still not sure. 

Can’t equal exactly, because the QEMU process itself has to consume some memory in addition to the VM’s memory

**Comment by deviantintegral** [comment]
I'm seeing something similar, but with `x86_64` (running on Intel). In my case, the VM is set to 16GB. I think large amounts of IO can trigger it. For example, I have may GB of Docker images exported, and when I load them with `docker load` by the time it's done `qemu-system-x86_64` will be using ~30GB of Real Memory.

**Reply by jdolitsky** [reply]
This might make a lot of sense (the IO). We are running a registry on the VM and publishing several images to it.
